services/src/datasets/external/natur40.rs

```diff
@@ -0,0 +1,290 @@
+        Ok(Box::new(Natur40DataProvider {
+            id: self.id,
+            base_url: self.base_url,
+            user: self.user,
+            password: self.password,
+        }))
+    }
+
+    fn type_name(&self) -> String {
+        "Natur4.0".to_owned()
+    }
+
+    fn name(&self) -> String {
+        self.name.clone()
+    }
+
+    fn id(&self) -> DatasetProviderId {
+        self.id
+    }
+}
+
+pub struct Natur40DataProvider {
+    id: DatasetProviderId,
+    base_url: String,
+    user: String,
+    password: String,
+}
+
+#[derive(Deserialize, Debug)]
+struct RasterDb {
+    name: String,
+    title: String,
+    #[serde(deserialize_with = "string_or_string_array")]
+    tags: Vec<String>,
+}
+
+impl RasterDb {
+    fn url_from_name(base_url: &str, name: &str) -> String {
+        format!(
+            "WCS:{}/rasterdb/{name}/wcs?VERSION=1.0.0&COVERAGE={name}",
+            base_url,
+            name = name
+        )
+    }
+
+    fn url(&self, base_url: &str) -> String {
+        Self::url_from_name(base_url, &self.name)
+    }
+}
+
+#[derive(Deserialize, Debug)]
+struct RasterDbs {
+    rasterdbs: Vec<RasterDb>,
+    #[serde(deserialize_with = "string_or_string_array")]
+    tags: Vec<String>,
+    session: String,
+}
+
+#[async_trait]
+impl DatasetProvider for Natur40DataProvider {
+    async fn list(&self, _options: Validated<DatasetListOptions>) -> Result<Vec<DatasetListing>> {
+        // TODO: query the other dbs as well
+        let raster_dbs = self.load_raster_dbs().await?;
+
+        let mut listing = vec![];
+
+        let datasets = raster_dbs
+            .rasterdbs
+            .iter()
+            .map(|db| self.load_dataset(db.url(&self.base_url)));
+        let datasets: Vec<Result<gdal::Dataset>> = join_all(datasets).await;
+
+        for (db, dataset) in raster_dbs.rasterdbs.iter().zip(&datasets) {
+            if let Ok(dataset) = dataset {
+                for band_index in 1..dataset.raster_count() {
+                    if let Ok(result_descriptor) =
+                        raster_descriptor_from_dataset(&dataset, band_index, None)
+                    {
+                        // TODO: get label from rasterband
+
+                        listing.push(Ok(DatasetListing {
+                            id: DatasetId::External(ExternalDatasetId {
+                                provider_id: self.id,
+                                dataset_id: format!("{}:{}", db.name.clone(), band_index),
+                            }),
+                            name: db.title.clone(),
+                            description: format!("Band: {}", band_index),
+                            tags: db.tags.clone(),
+                            source_operator: "GdalSource".to_owned(),
+                            result_descriptor: TypedResultDescriptor::Raster(result_descriptor),
+                            symbology: None, // TODO: build symbology
```

Comment:
Does Stephan provide infos about this?
If so, we should add at least a task to the story.

Reply:
no, we only get the infos about the dataset that I mapped in the `RasterDbs` type

Reply:
Maybe we (=Sohaib) can discuss this with Stephan if there is default colorization. This would mean either creating a task or just leaving the comment for later.

Reply:
https://trello.com/c/IlvLaYLo